model/src/test/kotlin/spdx/SpdxExpressionParserTest.kt

```diff
@@ -0,0 +1,188 @@
+
+                spdxExpression shouldBe SpdxCompoundExpression(
+                        SpdxLicenseIdExpression("license1", true),
+                        SpdxOperator.AND,
+                        SpdxCompoundExpression(
+                                SpdxCompoundExpression(
+                                        SpdxLicenseIdExpression("license2", false),
+                                        SpdxOperator.WITH,
+                                        SpdxLicenseExceptionExpression("exception1")
+                                ),
+                                SpdxOperator.OR,
+                                SpdxCompoundExpression(
+                                        SpdxLicenseIdExpression("license3", true),
+                                        SpdxOperator.AND,
+                                        SpdxCompoundExpression(
+                                                SpdxLicenseIdExpression("license4", false),
+                                                SpdxOperator.WITH,
+                                                SpdxLicenseExceptionExpression("exception2")
+                                        )
+                                )
+                        )
+                )
+            }
+
+            "bind + stronger than WITH" {
+                val spdxExpression = parseSpdxExpression("license+ WITH exception")
+
+                spdxExpression shouldBe SpdxCompoundExpression(
+                        SpdxLicenseIdExpression("license", true),
+                        SpdxOperator.WITH,
+                        SpdxLicenseExceptionExpression("exception")
+                )
+            }
+
+            "bind WITH stronger than AND" {
+                val spdxExpression = parseSpdxExpression("license1 AND license2 WITH exception")
+
+                spdxExpression shouldBe SpdxCompoundExpression(
+                        SpdxLicenseIdExpression("license1", false),
+                        SpdxOperator.AND,
+                        SpdxCompoundExpression(
+                                SpdxLicenseIdExpression("license2", false),
+                                SpdxOperator.WITH,
+                                SpdxLicenseExceptionExpression("exception")
+                        )
+                )
+            }
+
+            "bind AND stronger than OR" {
+                val spdxExpression = parseSpdxExpression("license1 OR license2 AND license3")
+
+                spdxExpression shouldBe SpdxCompoundExpression(
+                        SpdxLicenseIdExpression("license1", false),
+                        SpdxOperator.OR,
+                        SpdxCompoundExpression(
+                                SpdxLicenseIdExpression("license2", false),
+                                SpdxOperator.AND,
+                                SpdxLicenseIdExpression("license3", false)
+                        )
+                )
+            }
+
+            "bind the same operator in order" {
```

Comment:
how about: "and" should be "left associative" ?

Reply:
Done.